libraries/bot-applicationinsights/src/test/java/com/microsoft/bot/applicationinsights/BotTelemetryClientTests.java

```diff
@@ -0,0 +1,188 @@
+        properties.put("hello", "value");
+        metrics.put("metric", 0.6);
+
+        botTelemetryClient.trackAvailability(
+            "test",
+            OffsetDateTime.now(),
+            Duration.ofSeconds(1000), // TODO: use computer ticks
+            "run location",
+            true,
+            "message",
+            properties,
+            metrics);
+
+        Mockito.doAnswer(invocation -> {
+            AvailabilityTelemetry availabilityTelemetry = invocation.getArgument(0);
+
+            Assert.assertEquals("test", availabilityTelemetry.getName());
+            Assert.assertEquals("message", availabilityTelemetry.getData().getMessage());
+            Assert.assertEquals("value", availabilityTelemetry.getProperties().get("hello"));
+            Assert.assertEquals((Double) 0.6, availabilityTelemetry.getMetrics().get("metric"));
+
+            return null;
+        }).when(mockTelemetryChannel).send(Mockito.any(AvailabilityTelemetry.class));
+    }
+
+    @Test
+    public void trackEventTest() {
+        Map<String, String> properties = new HashMap<>();
+        properties.put("hello", "value");
+        Map<String, Double> metrics = new HashMap<>();
+        metrics.put("metric", 0.6);
+        
+        botTelemetryClient.trackEvent("test", properties, metrics);
+
+        Mockito.doAnswer(invocation -> {
+            EventTelemetry eventTelemetry = invocation.getArgument(0);
+
+            Assert.assertEquals("test", eventTelemetry.getName());
+            Assert.assertEquals("value", eventTelemetry.getProperties().get("hello"));
+            Assert.assertEquals((Double) 0.6, eventTelemetry.getMetrics().get("metric"));
+
+            return null;
+        }).when(mockTelemetryChannel).send(Mockito.any(EventTelemetry.class));
+    }
+
+    @Test
+    public void trackDependencyTest() {
+        botTelemetryClient.trackDependency(
+                "test", 
+                "target", 
+                "dependencyname", 
+                "data", 
+                OffsetDateTime.now(), 
+                Duration.ofSeconds(1000), // TODO: use computer ticks
+                "result", false);
+
+        Mockito.doAnswer(invocation -> {
+            RemoteDependencyTelemetry remoteDependencyTelemetry = invocation.getArgument(0);
+
+            Assert.assertEquals("test", remoteDependencyTelemetry.getType());
+            Assert.assertEquals("target", remoteDependencyTelemetry.getTarget());
+            Assert.assertEquals("dependencyname", remoteDependencyTelemetry.getName());
+            Assert.assertEquals("result", remoteDependencyTelemetry.getResultCode());
+            Assert.assertFalse(remoteDependencyTelemetry.getSuccess());
+
+            return null;
+        }).when(mockTelemetryChannel).send(Mockito.any(RemoteDependencyTelemetry.class));
+    }
+
+    @Test
+    public void trackExceptionTest() {
+        Exception expectedException = new Exception("test-exception");
+
```

Comment:
Are you missing to do `botTelemetryClient.trackException(...)` like [C#](https://github.com/microsoft/botbuilder-dotnet/blob/main/tests/Microsoft.Bot.Builder.ApplicationInsights.Tests/BotTelemetryClientTests.cs#L114)?